javalib/src/main/scala/java/lang/UnixProcess.scala

```diff
@@ -211,17 +218,21 @@ object UnixProcess {
   private val processes =
     scala.collection.mutable.HashMap.empty[CInt, UnixProcess]
   private def signalHandler(sig: CInt): Unit = {
-    val procInfo = stackalloc[ProcessMonitor.proc_info]
-    ProcessMonitor.lastProcInfo(procInfo)
-    val pid = !procInfo._1
-    processes get pid match {
-      case Some(process) =>
-        process.setExitValue(!procInfo._2)
-        processes -= pid
-        process.alertWaitingThreads()
-      case _ =>
+    if (ProcessMonitor.inGC != 0) {
```

Comment:
This doesn't look like a real solution, but just a workaround for the symptom of the problem.

Reply:
I agree that it is a workaround rather than a solution. At the same time, the entire monitoring implementation is a workaround not having threads. Since nothing is thread safe in scala native code, the monitor thread must communicate with the native thread using only data that isn't allocated by the gc. There is cleanup work on the process that must happen when the process exits that cannot be done on the monitor thread because it must modify scala native objects to set the exit value and cleanup the io streams. The existing hack is to interrupt the main thread to perform this work. This works well, except when GC is running.

As ugly as it is, I have found that this change does reliably make the processes work. I wrote a simple scala native project that more or less just called git status (and did some other work to help trigger garbage collection). Prior to this change, the script would crash roughly O(5%) of the time. After this change, the script ran 100000+ times without a crash.

I know that multithreading is still a ways out so I'm not relying on that at all. I'm just saying that whenever that is merged in, the process monitoring code should be re-implemented with scala native threads.